### PR TITLE
Add flake8

### PR DESCRIPTION
**Pull request by cschams**
Adds lint requirements and CircleCI configuration for flake8. Also reformats code to address issues raised.

### CODE REVIEW

File: flask_common/mongo/documents.py [hunk 1]
```diff
@@ -127,8 +127,8 @@ def delete(self, **kwargs):
             self.modify(set__is_deleted=self.is_deleted)
 
     @queryset_manager
-    def all_objects(doc_cls, queryset):
-        if not hasattr(doc_cls, '_all_objs_queryset'):
+    def all_objects(self, queryset):
```
Comment (tsx) [inline flask_common/mongo/documents.py]: This is a false positive imho. Queryset managers are not instances of documents. Maybe turn off this specific rule?

File: flask_common/utils/legacy.py [hunk 2]
```diff
@@ -697,3 +694,39 @@ def __getitem__(self, key):
         if self.data is None:
             self.data = list(self.f())
         return self.data[key]
+
+
+__all__ = [
```
Comment (tsx) [inline flask_common/utils/legacy.py]: I prefer `__all__` to be at the top of the file after imports but don't care too much.

File: flask_common/utils/legacy.py [hunk 3]
```diff
@@ -26,9 +26,6 @@
 from socket import gethostname
 
 
-from ..enum import Enum  # deprecated
```
Comment (tsx) [inline flask_common/utils/legacy.py]: Make sure our private code doesn't import Enum from here.

File: flask_common/mongo/documents.py (hunk 1 above)
Comment (cschams) [reply flask_common/mongo/documents.py]: Thanks for the suggestion.

File: flask_common/utils/legacy.py (hunk 3 above)
Comment (cschams) [reply flask_common/utils/legacy.py]: Did a code search, looks good: https://github.com/search?p=1&q=org%3Acloseio+flask_common+Enum&type=Code

File: flask_common/mongo/documents.py (hunk 1 above)
Comment (cschams) [reply flask_common/mongo/documents.py]: Looks like I turned off all name rules as recommended by black. I will undo this change.

Comment (tsx) [reply flask_common/mongo/documents.py]: I don't think black has anything to do with `self` naming, so it's just a weird mongoengine interface.

**Pull request merged**